### ISSUE TITLE
[extend] with useDefault=true doesn't work; only custom rules are matched

### ISSUE THREAD

**Issue by sbolel** [gitleaks/gitleaks#1164]
**Describe the bug**

When I create a `.gitleaks.toml` config and define a custom rule, it overrides the default ruleset and only the custom rules are detected. Adding the `[extend]` option with `useDefault` in `.gitleaks.toml` does not change that behavior.

If I stage the following file in git:

```
USER_POOL_ID="AKIAIOSFBBAA1EXAMPLE"
TEST_API="test-api-9bc40c3b-391f-487f-9ced-5da08af1481e"
```

And with the following `.gitleaks.toml` defined in the root:

```
[extend]
useDefault = true

[[rules]]
id = "test-api-token"
description = "Test API Tokens"
regex = '''test-api-[0-9a-f]{8}-[0-9a-f]{4}-4[0-9a-f]{3}-[8|9aA|bB][0-9a-f]{3}-[0-9a-f]{12}'''
tags = ["credentials","hardcoded","api"]
keywords = [
  "authorization",
  "api",
  "token",
]
```

I get the result:
```
Finding:     TEST_API="REDACTED"
Secret:      REDACTED
RuleID:      sbom-api-token
Entropy:     4.124100
File:        env
Line:        2
Fingerprint: env:sbom-api-token:2
```

However, when I remove that `.gitleaks.toml` file, I get the following result"

```
Finding:     USER_POOL_ID="REDACTED"
Secret:      REDACTED
RuleID:      aws-access-token
Entropy:     3.684184
File:        env
Line:        1
Fingerprint: env:aws-access-token:1
```

**To Reproduce**
- n/a

**Expected behavior**
- both the default rules and the custom defined rules should be matched when using a `.gitleaks.toml` config with `[extend]` `useDefault = true`.

**Screenshots**
- n/a

**Basic Info (please complete the following information):**
 - OS: macos 13.3.1
 - Gitleaks Version: 8.16.3

**Additional context**
- running through pre-commit and through the cli both result in the same behavior.

cc @zricethezav

**Comment by rgmz** [comment]
Do you see the following line if you enable `--log-level=debug`?

```
1:56PM DBG using gitleaks config .gitleaks.toml from `--config`
1:56PM DBG extending config with default config
```

I tested this with Gitleaks v8.16.3 on WSL (Ubuntu 22.04.2 LTS; Windows 10) and can't seem to reproduce the behaviour:

```bash
$ gitleaks detect --config .gitleaks.toml --verbose --log-level=debug --source ./gitleaks-1164/

    ○
    │╲
    │ ○
    ○ ░
    ░    gitleaks

1:58PM DBG using gitleaks config .gitleaks.toml from `--config`
1:58PM DBG extending config with default config
1:58PM DBG executing: /usr/bin/git -C gitleaks-1164 log -p -U0 --full-history --all
Finding:     TEST_API="test-api-9bc40c3b-391f-487f-9ced-5da08af1481e"
Secret:      test-api-9bc40c3b-391f-487f-9ced-5da08af1481e
RuleID:      test-api-token
Entropy:     4.079656
File:        env
Line:        2
Commit:      a58a0c7c68fbc37395bfbaded81758b29e8d6c61
Author:      Richard Gomez
Email:       rrichard.gomez@example.com
Date:        2023-04-25T17:58:27Z
Fingerprint: a58a0c7c68fbc37395bfbaded81758b29e8d6c61:env:test-api-token:2

Finding:     USER_POOL_ID="AKIAIOSFBBAA1EXAMPLE"
Secret:      AKIAIOSFBBAA1EXAMPLE
RuleID:      aws-access-token
Entropy:     3.441446
File:        env
Line:        1
Commit:      a58a0c7c68fbc37395bfbaded81758b29e8d6c61
Author:      Richard Gomez
Email:       rrichard.gomez@example.com
Date:        2023-04-25T17:58:27Z
Fingerprint: a58a0c7c68fbc37395bfbaded81758b29e8d6c61:env:aws-access-token:1

1:58PM INF 1 commits scanned.
1:58PM DBG Note: this number might be smaller than expected due to commits with no additions
1:58PM INF scan completed in 56.9ms
1:58PM WRN leaks found: 2
```

**Comment by sbolel** [comment]
Thanks for the quick reply! I made a total dummy mistake, and everything was working as intended.

I left out the most relevant part of the config I had -- `allowlist`.

```
[allowlist]
description = "global allowlist."
paths = [
  '''test/*''',
  '''.yarn/*'''
]
stopwords = ['''EXAMPLE''']
```

When I take out `stopwords`, I saw both secrets and that's when I realized I put the word "EXAMPLE" in the `USER_POOL_ID`. I added more tests and confirmed that everything is working as expected. 

The command I am running is `gitleaks protect --config .gitleaks.toml --verbose --staged` and yes, I do see that the extending is happening.

Thank you again! Closing this out.
